### ISSUE TITLE
inversify version clash

### ISSUE THREAD

**Issue by mindaugasdirg** [iTwin/object-storage#73]
**Problem**

Currently inversify is set as internal dependency in object-storage packages. This hides compatible versions of inversify from package clients.

If user installs a incompatible version of it will result in unhelpful error message. Here is an example:
Using newest version of inversify:
`npm install --save inversify@6.0.1 reflect-metadata`

code:
```ts
const serverBindings = new AzureServerStorageBindings();
serverBindings.register(container, {
    accountKey,
    accountName,
    baseUrl,
    dependencyName: "azure"
});
```

Error from build time:
```
error TS2345: Argument of type 'import("D:/imodel-transformations/imodel-transformations/agent/node_modules/inversify/lib/container/container").Container' is not assignable to parameter of type 'import("D:/imodel-transformations/imodel-transformations/packages/imodel-transformations-local/node_modules/inversify/dts/container/container").Container'.
  Types of property 'parent' are incompatible.
    Type 'import("D:/imodel-transformations/imodel-transformations/agent/node_modules/inversify/lib/interfaces/interfaces").interfaces.Container | null' is not assignable to type 'import("D:/imodel-transformations/imodel-transformations/packages/imodel-transformations-local/node_modules/inversify/dts/interfaces/interfaces").interfaces.Container | null'.
      Type 'import("D:/imodel-transformations/imodel-transformations/agent/node_modules/inversify/lib/interfaces/interfaces").interfaces.Container' is not assignable to type 'import("D:/imodel-transformations/imodel-transformations/packages/imodel-transformations-local/node_modules/inversify/dts/interfaces/interfaces").interfaces.Container'.
        The types of 'bind(...).to(...).inSingletonScope().when' are incompatible between these types.
          Type '(constraint: (request: Request) => boolean) => BindingOnSyntax<any>' is not assignable to type '(constraint: (request: Request) => boolean) => BindingOnSyntax<T>'.
            Types of parameters 'constraint' and 'constraint' are incompatible.
              Types of parameters 'request' and 'request' are incompatible.
                Type 'import("D:/imodel-transformations/imodel-transformations/agent/node_modules/inversify/lib/interfaces/interfaces").interfaces.Request' is not assignable to type 'import("D:/imodel-transformations/imodel-transformations/packages/imodel-transformations-local/node_modules/inversify/dts/interfaces/interfaces").interfaces.Request'.
                  The types of 'parentContext.setCurrentRequest' are incompatible between these types.
                    Type '(request: import("D:/imodel-transformations/imodel-transformations/agent/node_modules/inversify/lib/interfaces/interfaces").interfaces.Request) => void' is not assignable to type '(request: import("D:/imodel-transformations/imodel-transformations/packages/imodel-transformations-local/node_modules/inversify/dts/interfaces/interfaces").interfaces.Request) => void'.
                      Types of parameters 'request' and 'request' are incompatible.
                        Type 'import("D:/imodel-transformations/imodel-transformations/packages/imodel-transformations-local/node_modules/inversify/dts/interfaces/interfaces").interfaces.Request' is not assignable to type 'import("D:/imodel-transformations/imodel-transformations/agent/node_modules/inversify/lib/interfaces/interfaces").interfaces.Request'.
                          The types of 'parentContext.container' are incompatible between these types.
                            Type 'Container' is missing the following properties from type 'Container': rebindAsync, unbindAsync, unbindAllAsync, isCurrentBound, and 9 more.

87         serverBindings.register(container, {
```

Error message isn't clear and package user doesn't get any indication what version of inversify should be used.

**Proposed Solution**

Set inversify as peer dependency. This would notify package users about unresolved peer dependencies of object storage packages or incompatible versions of inversify between different packages.

If you want I can create PR for it.

**Comment by austeja-bentley** [comment]
Hello, in the recently released 2.0.0 version of object-storage we made `inversify` a peer with minimum version requirement of `6.0.1`.

**Comment by mindaugasdirg** [comment]
Thanks for the update.
